File: src/main/java/seremis/geninfusion/inventory/slot/GISlot.java
package seremis.geninfusion.inventory.slot;

import net.minecraft.inventory.IInventory;
import net.minecraft.inventory.Slot;
import net.minecraft.item.ItemStack;

public class GISlot extends Slot {

    private IInventory tile;

    public GISlot(IInventory inventory, int id, int x, int y) {
        super(inventory, id, x, y);
        this.tile = inventory;
    }

    @Override
    public boolean isItemValid(ItemStack itemStack) {
        return this.tile.isItemValidForSlot(this.slotNumber, itemStack);
    }

    @Override
    public int getSlotStackLimit() {
        ItemStack itemStack = tile.getStackInSlot(this.slotNumber);

        if(itemStack != null) {
            return Math.min(itemStack.getMaxStackSize(), tile.getInventoryStackLimit());
        }

        return tile.getInventoryStackLimit();
    }

}

File: src/main/java/seremis/geninfusion/inventory/slot/GISlot.scala
package seremis.geninfusion.inventory.slot

import net.minecraft.inventory.{IInventory, Slot}
import net.minecraft.item.ItemStack

class GISlot(inventory: IInventory, id: Int, x: Int, y: Int) extends Slot(inventory, id, x, y) {

    val tile: IInventory = inventory

    override def isItemValid(itemStack: ItemStack): Boolean = {
        tile.isItemValidForSlot(this.slotNumber, itemStack)
    }

    override def getSlotStackLimit: Int = {
        val itemStack = tile.getStackInSlot(this.slotNumber)
        if (itemStack != null) {
            return Math.min(itemStack.getMaxStackSize, tile.getInventoryStackLimit)
        }
        tile.getInventoryStackLimit
    }
}